pygame/math.py

```diff
@@ -0,0 +1,1575 @@
+        """Return object for an elementwise operation."""
+        return ElementwiseVector2Proxy(self)
+
+    def rotate(self, angle):
+        """rotates a vector by a given angle in degrees."""
+        x, y = _rotate_2d(self._x, self._y, angle)
+        return Vector2(x, y)
+
+    def rotate_ip(self, angle):
+        """rotates the vector by a given angle in degrees in place."""
+        new_vec = self.rotate(angle)
+        self.x = new_vec._x
+        self.y = new_vec._y
+
+    def angle_to(self, vec):
+        """calculates the minimum angle to a given vector in degrees."""
+        if not isinstance(vec, Vector2):
+            vec = Vector2(vec)
+        angle_self_rad = math.atan2(self._y, self._x)
+        angle_other_rad = math.atan2(vec.y, vec.x)
+        return math.degrees(angle_other_rad - angle_self_rad)
+
+    def as_polar(self):
+        """returns a tuple with radial distance and azimuthal angle."""
+        r = self.length()
+        angle = math.degrees(math.atan2(self._y, self._x))
+        return (r, angle)
+
+    def from_polar(self, polar):
+        """Sets x and y from a polar coordinates tuple."""
+        if isinstance(polar, tuple) and len(polar) == 2:
+            r = polar[0]
+            angle_rad = math.radians(polar[1])
+            self.x = r * math.cos(angle_rad)
+            self.y = r * math.sin(angle_rad)
+        else:
+            raise TypeError("Expected 2 element tuple (radius, angle), but got {}"
+                            .format(polar))
+
+
+class Vector3(object):
+    """A 3-Dimensional Vector."""
+
+    def __init__(self, *args):
+        if len(args) == 0:
+            self.x = 0.0
+            self.y = 0.0
+            self.z = 0.0
+        elif len(args) == 1:
+            if isinstance(args[0], Vector3):
+                self.x = args[0].x
+                self.y = args[0].y
+                self.z = args[0].z
+            elif isinstance(args[0], string_types):
+                if args[0].startswith("<Vector3(") and args[0].endswith(")>"):
+                    tokens = args[0][9:-2].split(",")
+                    try:
+                        self.x = float(tokens[0])
+                        self.y = float(tokens[1])
+                        self.z = float(tokens[2])
+                    except ValueError:
+                        raise TypeError("Invalid string arguments - not floats ({}).".
+                                        format(args[0]))
+                else:
+                    raise TypeError("Invalid string argument - not like __repr__ ({}).".
+                                    format(args[0]))
+            elif len(args[0]) == 3:
+                self.x = args[0][0]
+                self.y = args[0][1]
+                self.z = args[0][2]
+            else:
+                raise TypeError("Invalid argument length ({}).".format(len(args[0])))
+        else:
+            self.x = args[0]
+            self.y = args[1]
+            self.z = args[2]
+
```

Comment:
Likewise pygame supports initiliasing Vector3 with either 1 or 2 numbers. Again this should be supported or the discrepency should be documented.